airbyte-workers/src/main/java/io/airbyte/workers/tracing/TemporalSdkInterceptor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.tracing;
+
+import com.google.common.annotations.VisibleForTesting;
+import datadog.trace.api.interceptor.MutableSpan;
+import datadog.trace.api.interceptor.TraceInterceptor;
+import io.micronaut.core.util.CollectionUtils;
+import java.util.ArrayList;
+import java.util.Collection;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Custom {@link TraceInterceptor} to handle Temporal SDK traces that include a non-error used to
+ * exit Workflows.
+ */
+@Slf4j
+public class TemporalSdkInterceptor implements TraceInterceptor {
+
+  /**
+   * Trace resource name used to scope the filtering performed by this interceptor.
+   */
+  static final String CONNECTION_MANAGER_WORKFLOW_IMPL_RESOURCE_NAME = "ConnectionManagerWorkflowImpl.run";
+
+  /**
+   * Error message tag key name that contains the Temporal exit error message.
+   */
+  static final String ERROR_MESSAGE_TAG_KEY = "error.message";
+
+  /**
+   * Temporal exit error message text.
+   */
+  static final String EXIT_ERROR_MESSAGE = "exit";
+
+  @Override
+  public Collection<? extends MutableSpan> onTraceComplete(final Collection<? extends MutableSpan> trace) {
+    final var filtered = new ArrayList<MutableSpan>();
```

Comment:
Nit; we could use a kotlin/scala like syntax here by using val. It is not standard so feel free to ignore it.